google-cloud-spanner/src/test/java/com/google/cloud/spanner/XGoogSpannerRequestIdTest.java

```diff
@@ -157,40 +160,83 @@ private void assertMonotonicityOfIds(String prefix, List<XGoogSpannerRequestId>
               + String.join("\n\t", violations.toArray(new String[0])));
     }
 
-    public static class methodAndRequestId {
-      String method;
-      String requestId;
-
-      public methodAndRequestId(String method, String requestId) {
-        this.method = method;
-        this.requestId = requestId;
-      }
-
-      public String toString() {
-        return "{" + this.method + ":" + this.requestId + "}";
-      }
-    }
-
-    public methodAndRequestId[] accumulatedUnaryValues() {
-      List<methodAndRequestId> accumulated = new ArrayList();
+    public MethodAndRequestId[] accumulatedUnaryValues() {
+      List<MethodAndRequestId> accumulated = new ArrayList();
       this.unaryResults.forEach(
           (String method, CopyOnWriteArrayList<XGoogSpannerRequestId> values) -> {
             for (int i = 0; i < values.size(); i++) {
-              accumulated.add(new methodAndRequestId(method, values.get(i).toString()));
+              accumulated.add(new MethodAndRequestId(method, values.get(i)));
             }
           });
-      return accumulated.toArray(new methodAndRequestId[0]);
+      return accumulated.toArray(new MethodAndRequestId[0]);
     }
 
-    public methodAndRequestId[] accumulatedStreamingValues() {
-      List<methodAndRequestId> accumulated = new ArrayList();
+    public MethodAndRequestId[] accumulatedStreamingValues() {
+      List<MethodAndRequestId> accumulated = new ArrayList();
       this.streamingResults.forEach(
           (String method, CopyOnWriteArrayList<XGoogSpannerRequestId> values) -> {
             for (int i = 0; i < values.size(); i++) {
-              accumulated.add(new methodAndRequestId(method, values.get(i).toString()));
+              accumulated.add(new MethodAndRequestId(method, values.get(i)));
             }
           });
-      return accumulated.toArray(new methodAndRequestId[0]);
+      return accumulated.toArray(new MethodAndRequestId[0]);
+    }
+
+    public void checkExpectedUnaryXGoogRequestIds(MethodAndRequestId... wantUnaryValues) {
+      MethodAndRequestId[] gotUnaryValues = this.accumulatedUnaryValues();
+      sortValues(gotUnaryValues);
```

Comment:
Is it correct that we are sorting these accumulated values here? Doesn't that mean that if we receive the request-ids out of the expected order on the (mock) server, this sort action hides any potential test failures due to the request-ids having been out of order?